orchestra/tests/test_todos.py

```diff
@@ -161,11 +167,174 @@ def test_create_todo_with_due_datetime(self):
+            content_type='application/json')
+        updated_todolist_template = TodoListTemplateSerializer(
+            TodoListTemplate.objects.get(id=todolist_template.id)).data
+
+        self.assertEqual(resp.status_code, 200)
+        self._verify_todolist_template_content(
+            updated_todolist_template, self._todolist_template_data(
+                todolist_template.slug,
+                todolist_template.name,
+                updated_description))
+
+    def test_todolist_template_list_create(self):
+        self._verify_todolist_template_list([])
+        self._verify_todolist_template_creation()
+        self._verify_todolist_template_list(
+            [self._todolist_template_data(
+                self.todolist_template_slug,
+                self.todolist_template_name,
+                self.todolist_template_description)])
+
+    def _verify_todo_content(self, todo, expected_todo):
+        todo = dict(todo)
+        created_at = todo.pop('created_at')
+        todo_id = todo.pop('id')
+        self.assertEqual(todo, expected_todo)
+        self.assertGreater(len(created_at), 0)
+        self.assertGreaterEqual(todo_id, 0)
+
+    def test_add_todos_from_todolist_template_success(self):
+        num_todos = Todo.objects.all().count()
+        add_todos_from_todolist_template_url = \
+            reverse('orchestra:todos:add_todos_from_todolist_template')
+        todolist_template = TodoListTemplateFactory(
+            slug=self.todolist_template_slug,
+            name=self.todolist_template_name,
+            description=self.todolist_template_description,
+            todos={'items': [{
+                'id': 1,
+                'description': 'todo parent',
+                'items': [{
+                    'id': 2,
+                    'description': 'todo child',
+                    'items': []
+                }]
+            }]},
+        )
+        resp = self.request_client.post(
+            add_todos_from_todolist_template_url,
+            {
+                'todolist_template': todolist_template.id,
+                'task': self.task.id,
+            })
+
+        self.assertEqual(resp.status_code, 200)
+        self.assertEqual(Todo.objects.all().count(), num_todos + 3)
+        todos = load_encoded_json(resp.content)
+        expected_todos = [
+            _todo_data(self.task, 'todo child', False,
+                       template=todolist_template.id,
+                       parent_todo=todos[1]['id']),
+            _todo_data(self.task, 'todo parent', False,
+                       template=todolist_template.id,
+                       parent_todo=todos[2]['id']),
+            _todo_data(self.task, self.todolist_template_name,
+                       False, template=todolist_template.id),
+        ]
+        for todo, expected_todo in zip(todos, expected_todos):
+            self._verify_todo_content(todo, expected_todo)
+
+    def test_add_todos_from_todolist_template_forbidden(self):
```

Comment:
So this is forbidden because there is no task_id?

Reply:
@paopow Yes. This means you cannot add todo without an associated task.